### ISSUE TITLE
MatchData#named_captures: NoMethodError: undefined method `named_captures'

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1512]
`MatchData#named_captures` is available since Ruby 2.4. See https://github.com/ruby/ruby/blob/v2_4_0/NEWS

To reproduce:

```ruby
/(?<fname>.+) (?<lname>.+)/.match('John Doe').named_captures
```

Error
```
NoMethodError: undefined method `named_captures' for #<MatchData "John Doe" fname:"John" lname:"Doe">:MatchData
	from (irb):11
	from ~/.rubies/truffleruby-1.0.0-rc10/bin/irb:29:in `<main>'
```

**Comment by nirvdrum** [comment]
Thanks for bug report. This was added in 82b3f94096e0cc95202bfec8521d54b40812c71f.

**Comment by nirvdrum** [comment]
@deepj We released 1.0.0 RC 11 earlier today, which should include a fix for this issue. Please commont here if you see otherwise. Thanks again for the report!
